src/BenchmarkDotNet/Toolchains/Roslyn/Builder.cs

```diff
@@ -11,9 +12,72 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp;
 using OurPlatform = BenchmarkDotNet.Environments.Platform;
+using MonoRuntime = BenchmarkDotNet.Environments.MonoRuntime;
 
 namespace BenchmarkDotNet.Toolchains.Roslyn
 {
+    [PublicAPI]
+    public class MonoAotBuilder : Builder
+    {
+        MonoRuntime mono;
+
+        [PublicAPI]
+        public override BuildResult Build(GenerateResult generateResult, BuildPartition buildPartition, ILogger logger)
+        {
+            BuildResult result = base.Build (generateResult, buildPartition, logger);
+
+            if (!result.IsBuildSuccess)
+                return result;
+
+            var dllPath = generateResult.ArtifactsPaths.ExecutablePath;
+            var monoPath = mono.CustomPath ?? "mono";
+
+            try {
+                using (var process = new Process { StartInfo = CreateMonoAot(monoPath, dllPath) })
+                {
+                    logger.WriteLineInfo("// Execute: " + process.StartInfo.FileName + " " + process.StartInfo.Arguments);
+                    ConsoleHandler.Instance.SetProcess(process);
+                    process.Start();
+                    process.WaitForExit(); // should we add timeout here?
+
+                    if (process.ExitCode != 0)
+                        logger.WriteLineError("Attempt to AOT failed: " + process.StartInfo.FileName + " " + process.StartInfo.Arguments);
```

Comment:
**important**: in this case you should also:

1. log the output so the user knows why it has failed
2. return `BuildResult.Failure` instead of `result` from base builder which is success in this case

Reply:
1. The output logging request here confuses me. The `CreateMonoAot` function sets `RedirectStandardOutput = true`, so we should be seeing the output from this command in the stdout of the BDN runner. I'll add the error code though.

2. Will do